### PR TITLE
Add player_free_spawners

### PR DESCRIPTION
**Pull request by SFort**

### CODE REVIEW

**Comment by exaskye** [comment]
This is potentially dangerous on a normal world due to hidden spawners underground, and makes dungeons really, *really* dangerous pre-loaded monsterboxes. This is especially problematic in abandoned mineshafts, which are always open and can allow the spiders to roam free, outside the area the spawners check for capping.

I'll accept this patch, but I'm moving it to Situational. There's an idea on my backlog that makes the *countdown timer* tick even when no players are nearby, meaning that as soon as a player comes in range it will immediately spawn. I feel this is a better tradeoff.

File: src/main/java/com/unascribed/fabrication/mixin/f_balance/player_free_spawners/MixinMobSpawnerLogic.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+import org.spongepowered.asm.mixin.Mixin;
+import org.spongepowered.asm.mixin.injection.At;
+import org.spongepowered.asm.mixin.injection.Inject;
+import org.spongepowered.asm.mixin.injection.callback.CallbackInfoReturnable;
+
+@Mixin(MobSpawnerLogic.class)
+@EligibleIf(configEnabled="*.player_free_spawners")
+public class MixinMobSpawnerLogic {
+	
+	@Inject(method="isPlayerInRange()Z", at=@At(value="HEAD", target="Lnet/minecraft/world/MobSpawnerLogic;update()V"), cancellable=true)
```
Comment (exaskye) [inline src/main/java/com/unascribed/fabrication/mixin/f_balance/player_free_spawners/MixinMobSpawnerLogic.java:15]: One note; `target=` on a HEAD doesn't do anything.

File: src/main/java/com/unascribed/fabrication/mixin/f_balance/player_free_spawners/MixinMobSpawnerLogic.java [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+	public void isPlayerInRange(CallbackInfoReturnable<Boolean> cir) {
+		if (MixinConfigPlugin.isEnabled("*.player_free_spawners")) {
+			cir.setReturnValue(true);
+			cir.cancel();
```
Comment (exaskye) [inline src/main/java/com/unascribed/fabrication/mixin/f_balance/player_free_spawners/MixinMobSpawnerLogic.java:19]: Er, one more note. Calling `setReturnValue` internally calls `cancel`, so this line is a no-op.

**Comment by SFort** [comment]
> really dangerous pre-loaded monsterboxes

that's the idea :)
I've played with this for about a few months and it's fun

**Comment by exaskye** [comment]
For clarity, my initial usage of "dangerous" in my comment refers to lag concerns. I do like the idea of pre-loaded dungeons but mobs wandering out of their range is a concern.

**Pull request merged**